signalr-go-server/pkg/signalr/hubconnectionbase.go

```diff
@@ -0,0 +1,99 @@
+package signalr
+
+import (
+	"bytes"
+	"io"
+	"sync/atomic"
+)
+
+type HubConnectionBase struct {
```

Comment:
The layering I'd prefer would be to expose a lower level connection object and a concrete HubConnection that sits on top and stays internal to the package.

Reply:
HubConnection will become hubConnection, so this will also made internal. 
Or did you mean this: renaming HubConnectionBase to hubConnection, then introducing a 
`func newHubConnection(connection transport) hubConnection`
where transport is both a reader and writer and assigned to the hubConnections reader and writer members?

Reply:
I made a proposal in [31e00c3](https://github.com/davidfowl/signalr-ports/pull/6/commits/31e00c336bf26b86b325402b88bf9fb60d0788b9)